### ISSUE TITLE
catch operator regression

### ISSUE THREAD

**Issue by jinroh** [ReactiveX/RxJS#887]
Observables returned by the `catch` operator are not properly unsubscribed when used as inner-subscriptions.

Here is a small repro-code:

```js
Observable.interval(2000)
  .switchMap((i) => {
    return Observable
      .interval(200).do((j) => console.log("inner", i + "_" + j))
      .catch(() => {}); // <- notice this catch
  })
  .subscribe()
```

```
inner 0_0
inner 0_1
inner 0_2
inner 0_3
inner 0_4
inner 0_5
inner 0_6
inner 0_7
inner 0_8
inner 0_9
inner 1_0
inner 0_10
inner 1_1
inner 0_11
inner 1_2
inner 0_12
inner 1_3
inner 0_13
...
```
 It looks like it has been introduced in this commit 280f7eda0385d0dfca7cea474d46f8a6a8f07196.

**Comment by staltz** [comment]
And fixed by #886 .

**Comment by jinroh** [comment]
Haha ! I love this project. Thanks @staltz !
